pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -314,18 +317,24 @@ func getCertsFromSecretBundle(ctx context.Context, secretBundle kv.SecretBundle,
 }
 
 // Based on https://github.com/sigstore/sigstore/blob/8b208f7d608b80a7982b2a66358b8333b1eec542/pkg/signature/kms/azure/client.go#L258
-func getKeyFromKeyBundle(keyBundle kv.KeyBundle) (crypto.PublicKey, error) {
+func getKeyFromKeyBundle(keyBundle azkeys.KeyBundle) (crypto.PublicKey, error) {
 	webKey := keyBundle.Key
 	if webKey == nil {
 		return nil, re.ErrorCodeKeyInvalid.NewError(re.KeyManagementProvider, ProviderName, re.EmptyLink, nil, "found invalid key bundle, key must not be nil", re.HideStackTrace)
 	}
 
-	keyType := webKey.Kty
+	if webKey.Kty == nil {
+		return nil, re.ErrorCodeKeyInvalid.NewError(re.KeyManagementProvider, ProviderName, re.EmptyLink, nil, "found invalid key bundle, keytype must not be nil", re.HideStackTrace)
+	}
+
+	keyType := *webKey.Kty
 	switch keyType {
-	case kv.ECHSM:
-		webKey.Kty = kv.EC
-	case kv.RSAHSM:
-		webKey.Kty = kv.RSA
+	case azkeys.JSONWebKeyTypeECHSM:
```

Comment:
have we verified the new values for each case exact the existing values? Just want to double check

Reply:
Here is how the new consts are defined in azkeys:
```
const (
	JSONWebKeyTypeEC JSONWebKeyType = "EC"
	JSONWebKeyTypeECHSM JSONWebKeyType = "EC-HSM"
	JSONWebKeyTypeOct JSONWebKeyType = "oct"
	JSONWebKeyTypeOctHSM JSONWebKeyType = "oct-HSM"
	JSONWebKeyTypeRSA JSONWebKeyType = "RSA"
	JSONWebKeyTypeRSAHSM JSONWebKeyType = "RSA-HSM"
)
```

And here is how they were defined in keyvault (kv):
```
const (
	EC = "EC"
	ECHSM = "EC-HSM"
	Oct = "oct"
	RSA = "RSA"
	RSAHSM = "RSA-HSM"
)
```

I replicated the same two conditions ( both in the condition and the body of the if statements). Is that whatever you are concerned about or something else?